apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -84,7 +86,7 @@ const initialState: AichatState = {
 export const updateAiCustomization = createAsyncThunk(
   'aichat/updateAiCustomization',
   async (_, thunkAPI) => {
-    const state = thunkAPI.getState() as {lab: LabState; aichat: AichatState};
+    const state = thunkAPI.getState() as RootState;
```

Comment:
Do you mind explaining how `RootState` is assigned correctly here?

Reply:
Yep, `RootState` comes from here: https://github.com/code-dot-org/code-dot-org/blob/95f488ffef1fe252a86552c4b3586d7c8fa3f0ab/apps/src/types/redux.ts#L23

This describes the type of the global state object, for all redux slices that have been migrated to typescript. In theory if all of our redux code was in TypeScript, this RootState type would completely describe the global state, but for now we can use this in TypeScript redux files at least. Also either way is fine really, I was just experimenting with a way to type the `thunkAPI` implicitly, but found that the redux-toolkit docs recommend this approach: https://redux-toolkit.js.org/usage/usage-with-typescript#typing-async-thunks-inside-createslice

Reply:
Thanks so much for the explanation! Still wrapping my head around redux-toolkit.